pkg/controller/namespace_helper.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2023 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/controller/utils"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func NewSingleTenantNamespaceHelper(ns string) NamespaceHelper {
+	return &namespacer{
+		multiTenant:           false,
+		singleTenantNamespace: ns,
+	}
+}
+
+func NewNamespaceHelper(mt bool, singleTenantNS, multiTenantNS string) NamespaceHelper {
+	return &namespacer{
+		multiTenant:           mt,
+		singleTenantNamespace: singleTenantNS,
+		multiTenantNamespace:  multiTenantNS,
+	}
+}
+
+type NamespaceHelper interface {
+	// InstallNamespace returns the namespace that components will be installed into.
+	// for single-tenant clusters, this is tigera-manager. For multi-tenancy, this
+	// will be the tenant's namespace.
+	InstallNamespace() string
+
+	// TruthNamespace returns the namespace to use as the source of truth for storing data.
+	// For single-tenant installs, this is the tigera-operator namespace.
+	// For multi-tenant installs, this is tenant's namespace.
+	TruthNamespace() string
+
+	// BothNamespaces returns both the truth namespace and the install namespace.
+	BothNamespaces() []string
```

Comment:
It is used twice, the name is ambiguous. I think the reader of the code will be much better off if this would not exist, so it does not have to read this description.

Reply:
I guess so. It will be used in more places when we roll out multi-tenancy to more subsystems, and avoids reproducing some boilerplate. I'll get rid of it for now anyway.